platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+###### Generating a SignatureSchema
+```java
+SignatureSchema signatureSchema = SignatureSchema.from(Curve.ALT_BN128, GroupAssignament.G1_PUBLIC_KEYS);
+```
+###### Generating a Private Key
+```java
+   PairingPrivateKey pk = PairingPrivateKey.create(signatureSchema, new SecureRandom());
+   
+```
+###### Generating a Public Key
+```java
+   PairingPublicKey pbk = PairingPublicKey.create(pk);
+   
+```
+###### Generating a Signature
+```java
+    byte[] message = new byte[]{};
+    PairingSignature s = PairingSignature.sign(pk, message );
+   
+```
+###### Verifying a Signature
+```java
+static{
+    s.verify(pbk, message);
+}
+```
+##### Constraints
+This module will not depend on hedera-services artifacts, so it cannot include logging, metrics, configuration, or any other helper module from that repo.
+##### Dependencies
+swirlds-cryptography-pairings-API and runtime implementation
+##### Other considerations
+Some unknowns are worth investigating in a follow-up task:
+
+#### Swirlds Crypto TSS Library
+##### Overview
+This library implements the Groth21 TSS-specific primitives.
+
+##### Constraints
+Once built, this library will depend on artifacts constructed in other repositories.
+This library would accept Integer.MAX_VALUE -1 participants.
+
+##### Public API:
+###### `TssService`
+**Description**: This class handles all tss specific operations.
+
+**Link**: [TssService.java](tss%2FTssService.java)
+
+###### `TssParticipantDirectory`
+**Description**: This class holds all information about the participants in the scheme. Including: participants' EC public keys, public shares, private shares, number of shares. 
+
+**Link**: [TssParticipantDirectory.java](tss%2FTssParticipantDirectory.java)
+
+###### `TssMessage`
+**Description**: This class is used in the exchange of secret information between participants. Contains an encrypted message, a polynomial commitment, and a cryptographic proof that can be used to validate this message.
+
+**Link**:[TssMessage.java](tss%2FTssMessage.java)
+
+###### `TssShareId`
+**Description**: This class represents the unique identification of a share in the system. It a FieldElement, so it can be used as input in the polynomial.
+
+**Link**:[TssShareId.java](tss%2FTssShareId.java)
+
+###### `TssPrivateShare`
+**Description**: A record that contains a share ID, and the corresponding private key.
+
+**Link**:[TssPrivateShare.java](tss%2FTssPrivateShare.java)
+
+###### `TssPublicShare`
+**Description**: A record that contains a share ID, and the corresponding public key.
+
+**Link**:[TssPublicShare.java](tss%2FTssPublicShare.java)
+
+###### `TssShareSignature`
+**Description**: Represents a signature created from a TSSPrivateShare.
+
+**Link**:[TssShareSignature.java](tss%2FTssShareSignature.java)
+
+##### Examples:
+###### 1. Bootstrapping
+```java
+   TssService service = new TssService(signatureScheme, new Random());
+   PairingPrivateKey persistentParticipantKey = Requirements.loadECPrivateKey();
```

Comment:
Is this always singular?  Could this also be a list, like the public keys on the next line?

Reply:
The private key is unique per participant (Node).
The public keys are all the public keys for all the participants.

e.g.
In a 3-node setup, node 1 will have its privateKey and a list of 3 publicKeys.